src/frontend/data/stores/useParticipantWorkflow/index.ts

```diff
@@ -38,14 +43,21 @@ export const useParticipantWorkflow = create<State>((set, get) => ({
       asked: false,
       accepted: false,
       rejected: true,
+      usernameAlreadyExisting: false,
       kicked: false,
     });
   },
+  setUsernameAlreadyExisting: () => {
```

Comment:
To be fully correct`setUsernameAlreadyExisting` should set `asked` to false right?
As user asks to join discussion but its request is canceled because its username already exists. Then he can ask again to join with a new username?

Reply:
I decided to not set `asked` to false because the request asking to join the discussion is still available and the form is display again with an error message asking to choose an other username. The user can then choose a new one or abort by clicking the cancel button (ans ask is set to false at this time). If he decides to pick a new username, a new attempt to connect to the chat is made and if successful then the request to join the discussion is sent.